interceptors/protovalidate/protovalidate.go

```diff
@@ -41,55 +45,40 @@ func UnaryServerInterceptor(validator *protovalidate.Validator, opts ...Option)
 
 // StreamServerInterceptor returns a new streaming server interceptor that validates incoming messages.
 func StreamServerInterceptor(validator *protovalidate.Validator, opts ...Option) grpc.StreamServerInterceptor {
+	o := evaluateOpts(opts)
+
 	return func(
 		srv interface{},
 		stream grpc.ServerStream,
 		info *grpc.StreamServerInfo,
 		handler grpc.StreamHandler,
 	) error {
-		ctx := stream.Context()
-
-		wrapped := wrapServerStream(stream)
-		wrapped.wrappedContext = ctx
-		wrapped.validator = validator
-		wrapped.options = evaluateOpts(opts)
+		wrapped := wrapServerStream(stream, validator, o)
 
 		return handler(srv, wrapped)
 	}
 }
 
 func (w *wrappedServerStream) RecvMsg(m interface{}) error {
-	if err := w.ServerStream.RecvMsg(m); err != nil {
+	if err := validateMessage(w.validator, w.options, m); err != nil {
 		return err
 	}
 
-	msg := m.(proto.Message)
-	if w.options.shouldIgnoreMessage(msg.ProtoReflect().Type()) {
-		return nil
-	}
-	if err := w.validator.Validate(msg); err != nil {
-		return status.Error(codes.InvalidArgument, err.Error())
-	}
-
-	return nil
+	return w.ServerStream.RecvMsg(m)
```

Comment:
Is this correct? From what I can tell, `RecvMsg` is unmarshalling data into `m`, so we should validate _after_ calling it.